### PR TITLE
Fixed extra distance in scroll automation (closes #987)

### LINKED ISSUE #987: TestCafe scroll to target element incorrectly
### Are you requesting a feature or reporting a bug?

bug

### What is the current behavior?

TestCafe doesn't scroll container to show target element fully while performing an action on it

### What is the expected behavior?

TestCafe should scroll container to show element fully, when it possible (element is smaller, than container)

### How would you reproduce the current behavior (if this is a bug)?

Run test

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:
```html
<html>
<head>
    <title>Index</title>
    <style>
        .row {
            height: 24px;
            width: 75px;
            background: grey;
            border: 1px solid black;
        }
    </style>
</head>
<body>
<div id="verticalContainer" style="overflow-y:scroll; left: 100px; position: absolute; height:70px; width:100px">
    <div id="targetTop" class="row"></div>
    <div class="row"></div>
    <div id="targetBottom" class="row"></div>
</div>
</body>
</html>
```
Test code

```js
const verticalContainer   = Selector('#verticalContainer');

test('Scroll a vertical container to the bottom element', async t => {
    await t.click('#targetBottom');

    expect(await verticalContainer.scrollTop).to.be.least(8);
});

```

### Specify your

* operating system: Win10 x64
* testcafe version: 0.11.0-alpha
* node.js version: v6.9.1

### PR DESCRIPTION
**Pull request by georgiy-abbasov**
\cc @AlexanderMoskovkin, @helen-dikareva, @MarinaRukavitsyna

### CODE REVIEW

**Comment by georgiy-abbasov** [comment]
ping @helen-dikareva, @MarinaRukavitsyna

File: test/functional/fixtures/regression/gh-987/test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+describe('[Regression](GH-987)', function () {
+    describe('Should show the target element fully while performing an action', function () {
```
Comment (MarinaRukavitsyna) [inline test/functional/fixtures/regression/gh-987/test.js]: describe('Should fully show the target element on performing an action', function () {

File: test/functional/fixtures/regression/gh-987/testcafe-fixtures/index.test.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+
+fixture `Scroll a div container`
+    .page `http://localhost:3000/fixtures/regression/gh-987/pages/index.html`;
+
+
+const horizontalContainer = Selector('#horizontalContainer');
+const verticalContainer   = Selector('#verticalContainer');
+
+test('Scroll a vertical container to the bottom element', async t => {
```
Comment (MarinaRukavitsyna) [inline test/functional/fixtures/regression/gh-987/testcafe-fixtures/index.test.js]: test('Scroll the vertical container to the bottom element', async t => {

File: test/functional/fixtures/regression/gh-987/testcafe-fixtures/index.test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    await t.click('#targetBottom');
+
+    expect(await verticalContainer.scrollTop).to.be.least(8);
+});
+
+test('Scroll a vertical container to the top element', async t => {
```
Comment (MarinaRukavitsyna) [inline test/functional/fixtures/regression/gh-987/testcafe-fixtures/index.test.js]: test('Scroll the vertical container to the top element', async t => {

File: test/functional/fixtures/regression/gh-987/testcafe-fixtures/index.test.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    await t.eval(() => {
+        document.querySelector('#verticalContainer').scrollTop = 8;
+    });
+
+    await t.click('#targetTop');
+
+    expect(await verticalContainer.scrollTop).eql(0);
+});
+
+test('Scroll a horizontal container to the left element', async t => {
```
Comment (MarinaRukavitsyna) [inline test/functional/fixtures/regression/gh-987/testcafe-fixtures/index.test.js]: test('Scroll the horizontal container to the left element', async t => {

File: test/functional/fixtures/regression/gh-987/testcafe-fixtures/index.test.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    await t.eval(() => {
+        document.querySelector('#horizontalContainer').scrollLeft = 8;
+    });
+
+    await t.click('#targetLeft');
+
+    expect(await horizontalContainer.scrollLeft).eql(0);
+});
+
+test('Scroll a horizontal container to the right element', async t => {
```
Comment (MarinaRukavitsyna) [inline test/functional/fixtures/regression/gh-987/testcafe-fixtures/index.test.js]: test('Scroll the horizontal container to the right element', async t => {

**Comment by georgiy-abbasov** [comment]
FPR

**Pull request merged**